src/qml/BarcodeReader.qml

```diff
@@ -0,0 +1,267 @@
+          text: qsTr('Code Reader')
+          font: Theme.strongFont
+          color: Theme.mainColor
+          horizontalAlignment: Text.AlignHCenter
+          wrapMode: Text.WordWrap
+        }
+
+        QfToolButton {
+          id: closeButton
+          Layout.rightMargin: 10
+          Layout.alignment: Qt.AlignVCenter
+          iconSource: Theme.getThemeIcon( 'ic_close_black_24dp' )
+          bgcolor: "transparent"
+
+          onClicked: {
+            barcodeReader.close();
+          }
+        }
+      }
+    }
+
+    ColumnLayout {
+      width: parent.width
+      height: parent.height
+
+      Rectangle {
+        Layout.fillWidth: true
+        Layout.fillHeight: true
+
+        color: "#333333"
+        radius: 10
+        clip: true
+
+        VideoOutput {
+          anchors.fill: parent
+          anchors.margins: 6
+
+          source: camera
+          autoOrientation: true
+          fillMode: VideoOutput.PreserveAspectCrop
+
+          filters: [
+            BarcodeVideoFilter {
+              active: barcodeReader.visible
+              decoder: barcodeDecoder
+            }
+          ]
+        }
+
+        Rectangle {
+          id: decodedFlash
+          anchors.fill: parent
+          anchors.margins: 6
+
+          color: "transparent"
+          SequentialAnimation {
+            id: decodedFlashAnimation
+            PropertyAnimation { target: decodedFlash; property: "color"; to: "white"; duration: 0 }
+            PropertyAnimation { target: decodedFlash; property: "color"; to: "transparent"; duration: 500 }
+          }
+        }
+
+        Shape {
+          id: frame
+          anchors.fill: parent
+
+          ShapePath {
+            strokeWidth: 2.5
+            strokeColor: "#333333"
+            strokeStyle: ShapePath.SolidLine
+            joinStyle: ShapePath.MiterJoin
+            fillColor: "transparent"
+
+            startX: 5
+            startY: 10
+            PathArc { x: 10; y: 5; radiusX: 5; radiusY: 5 }
+            PathLine { x: frame.width - 10; y: 5; }
+            PathArc { x: frame.width - 5; y: 10; radiusX: 5; radiusY: 5 }
+            PathLine { x: frame.width - 5; y: frame.height - 10; }
+            PathArc { x: frame.width - 10; y: frame.height - 5; radiusX: 5; radiusY: 5 }
+            PathLine { x: 10; y: frame.height - 5; }
+            PathArc { x: 5; y: frame.height - 10; radiusX: 5; radiusY: 5 }
+            PathLine { x: 5; y: 10 }
+          }
+        }
+
+        Shape {
+          id: aim
+          anchors.fill: parent
+
+          ShapePath {
+            strokeWidth: 2.5
+            strokeColor: "white"
+            strokeStyle: ShapePath.SolidLine
+            joinStyle: ShapePath.MiterJoin
+            fillColor: "transparent"
+
+            startX: 20
+            startY: 60
+            PathLine { x: 20; y: 25 }
+            PathArc { x: 25; y: 20; radiusX: 5; radiusY: 5 }
+            PathLine { x: 60; y: 20 }
```

Comment:
I hope we never have to change these values, as we have to recalculate them manually. Other option would be to make them dynamically calculated like:
```
ShapePath {
    startX: 20
    startY: 60
    property int borderRadius:  5
    PathLine { x: parent.startX; y: parent.startX + parent.borderRadius }
    // ...
 }
```
No strong opinion here.